stacks_queues/queue.py

```diff
@@ -23,40 +24,103 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+
+        if (self._front == 0 and self.rear == self.buffer_size - 1) or self.rear == (self._front - 1):
+            raise QueueFullException
+        # queue is empty
+        elif (self._front == -1):
+            self.store_size += 1
+            self._front = 0
+            self.rear = 0
+            self.store[self.rear] = element
+        # wrap rear
+        elif (self.rear == self.buffer_size - 1):
+            self.store_size += 1
+            self.rear = 0
+            self.store[self.rear] = element
+        # rear is not at end of list
+        else:
+            self.store_size += 1
+            self.rear = self.rear + 1
+            self.store[self.rear] = element
```

Comment:
This can be made a bit simpler.

```suggestion
        else:
            self.store_size += 1
            self.rear = (self.rear + 1) % self.buffer_size
            self.store[self.rear] = element
```